src/event/metric.rs

```diff
@@ -360,6 +379,112 @@ impl Display for Metric {
+                Ok(())
+            }
+            [Segment::Field(namespace)] if namespace.as_str() == "namespace" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                self.namespace = Some(String::from_utf8_lossy(value).into_owned());
+                Ok(())
+            }
+            [Segment::Field(timestamp)] if timestamp.as_str() == "timestamp" => {
+                let value = value.as_timestamp().ok_or("value must be a timestamp")?;
+                self.timestamp = Some(*value);
+                Ok(())
+            }
+            [Segment::Field(kind)] if kind.as_str() == "kind" => {
+                let value = value.as_bytes().ok_or("value must be a string")?;
+                match String::from_utf8_lossy(value).as_ref() {
+                    "absolute" => {
+                        self.kind = MetricKind::Absolute;
+                        Ok(())
+                    }
+                    "incremental" => {
+                        self.kind = MetricKind::Incremental;
+                        Ok(())
+                    }
+                    _ => Err("metric kind must be `absolute` or `incremental`".into()),
+                }
+            }
+            _ => Err("invalid path".into()),
+        }
+    }
+
+    fn get(&self, path: &remap::Path) -> Result<Option<remap::Value>, String> {
+        // TODO construct a map of the whole metric.
+        if path.is_root() {
+            return Err("cannot set root path for a metric event".into());
+        }
+
+        match path.segments() {
+            [Segment::Field(name)] if name.as_str() == "name" => Ok(Some(self.name.clone().into())),
+            [Segment::Field(namespace)] if namespace.as_str() == "namespace" => {
+                Ok(self.namespace.clone().map(Into::into))
+            }
+            [Segment::Field(timestamp)] if timestamp.as_str() == "timestamp" => {
+                Ok(self.timestamp.map(Into::into))
+            }
+            [Segment::Field(kind)] if kind.as_str() == "kind" => Ok(Some(
+                match self.kind {
+                    MetricKind::Absolute => "absolute",
+                    MetricKind::Incremental => "incremental",
+                }
+                .into(),
+            )),
+            [Segment::Field(tags), Segment::Field(field)] if tags.as_str() == "tags" => {
+                Ok(self.tag_value(field.as_str()).map(|value| value.into()))
+            }
+            _ => Err("invalid path".into()),
+        }
+    }
+
+    fn paths(&self) -> Result<Vec<remap::Path>, String> {
+        Ok(["name", "namespace", "timestamp", "tags"]
+            .iter()
+            .map(|path| Path::from_str(path).expect("invalid path"))
+            .collect())
+    }
+
+    fn remove(&mut self, path: &remap::Path, _compact: bool) -> Result<(), String> {
+        if path.is_root() {
+            return Err("cannot set root path for a metric event".into());
```

Comment:
s/set/remove (this would be its own enum variant if we move to a snafu-derived error object)